Code/Mods/UI/Camera.cs

```diff
@@ -38,11 +40,16 @@ public void StartAimCoroutine(MonoBehaviour owner, bool isEnteringAimMode)
         {
             // Choose zoom in or out
             float targetZoom = 0f;
-            Func<bool> test = () => _zoomControlAmount.Value < targetZoom + AIM_COROUTINE_MARGIN;
+            if (_cachedZoom != null)
+                targetZoom = (float)_cachedZoom;
+                
+
+            Func<bool> test;
+            test = () => Mathf.Abs(_zoomControlAmount.Value) < Mathf.Abs(targetZoom) + AIM_COROUTINE_MARGIN;
             if (isEnteringAimMode)
             {
-                targetZoom = 1f;
-                test = () => _zoomControlAmount.Value > targetZoom - AIM_COROUTINE_MARGIN;
+                targetZoom = _aimZoomAmount;
+                test = () => Mathf.Abs(_zoomControlAmount.Value) > Mathf.Abs(targetZoom) + AIM_COROUTINE_MARGIN;
             }
```

Comment:
This whole block can be refactored as follow:

```csharp
var targetZoom = isEnteringAimMode ? _aimZoomAmount : _cachedZoom ?? 0f;
var test = () => Mathf.Abs(_zoomControlAmount.Value) < Mathf.Abs(targetZoom) + AIM_COROUTINE_MARGIN;
```
The lambda test is the same regardless of "isEnteringAimMode"
We can skip casting a float to bool to increase readability as well.
I have not test run this code but I think it will work.

Reply:
The lambda test is reversed if IsEnteringAimMode, one checks if it smaller, one checks if it larger, depending which way it wants to push it.